api/db/migrations/20250709083801_add-version-col-to-certificaiton-frameworks-challenges.js

```diff
@@ -0,0 +1,24 @@
+const TABLE_NAME = 'certification-frameworks-challenges';
+const COLUMN_NAME = 'version';
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+const up = async function (knex) {
+  await knex.schema.table(TABLE_NAME, function (table) {
+    table.string(COLUMN_NAME).comment('version of consolidated framework');
```

Comment:
Je propose d'expliquer un peu mieux nos données car il y a une règle implicite sur cette colonne.

```suggestion
    table.string(COLUMN_NAME).comment('version of consolidated framework. The version is by framework key, meaning that different frameworks can have the same version.');
```

Reply:
Une idee : autodocumenter nos données via une contrainte car notre decision est quand meme une regle pivot de cette table (il n'existe pour un framework données, une unicite sur le triplet key-challenge-version)

```sql
ALTER TABLE "certification-frameworks-challenges" ADD UNIQUE (key, version, challengeId)
```

Reply:
Nous pourrons peut-être refaire une PR avec ces infos ?

Il faut en faire une nouvelle après de toute façon pour la contrainte « not null » de la version. Je prépare le ticket avec tous ces points dedans !

https://1024pix.atlassian.net/browse/PIX-18680